health-services/resource-estimation-service/src/main/java/org/egov/processor/util/CampaignIntegrationUtil.java

```diff
@@ -0,0 +1,287 @@
+import org.egov.processor.web.models.ResourceMapping;
+import org.egov.processor.web.models.campaignManager.Boundary;
+import org.egov.processor.web.models.campaignManager.CampaignDetails;
+import org.egov.processor.web.models.campaignManager.CampaignRequest;
+import org.egov.processor.web.models.campaignManager.CampaignResources;
+import org.egov.processor.web.models.campaignManager.CampaignResponse;
+import org.egov.processor.web.models.campaignManager.CampaignSearchRequest;
+import org.egov.tracer.model.CustomException;
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Component
+@Slf4j
+public class CampaignIntegrationUtil {
+
+	private ServiceRequestRepository serviceRequestRepository;
+	private Configuration config;
+	private ObjectMapper mapper;
+
+	public CampaignIntegrationUtil(ServiceRequestRepository serviceRequestRepository, Configuration config,
+			ObjectMapper mapper, FilestoreUtil filestoreUtil, ParsingUtil parsingUtil) {
+
+		this.serviceRequestRepository = serviceRequestRepository;
+		this.config = config;
+		this.mapper = mapper;
+	}
+
+	/**
+	 * Updates campaign details based on the provided plan configuration request and response data.
+	 * This method integrates the campaign details obtained from the response into the provided plan configuration request.
+	 * It also updates the campaign boundaries and resources accordingly.
+	 *
+	 * @param planConfigurationRequest The plan configuration request containing the execution plan details.
+	 * @param response The response object containing campaign details.
+	 * @param campaignBoundaryList The list of campaign boundaries.
+	 * @param campaignResourcesList The list of campaign resources.
+	 */
+	public void updateCampaignDetails(PlanConfigurationRequest planConfigurationRequest,Object response,List<Boundary> campaignBoundaryList,List<CampaignResources> campaignResourcesList) {
+		CampaignResponse campaignResponse = null;
+		try {
+			campaignResponse = mapper.convertValue(response, CampaignResponse.class);
+			campaignResponse.getCampaign().get(0).setResources(campaignResourcesList);
+			Boundary[] array = campaignBoundaryList.toArray(new Boundary[0]);
+			campaignResponse.getCampaign().get(0).setBoundaries(campaignBoundaryList.toArray(new Boundary[0]));
+			serviceRequestRepository.fetchResult(
+					new StringBuilder(config.getProjectFactoryHostEndPoint() + config.getCampaignIntegrationUpdateEndPoint()),
+					buildCampaignRequestForUpdate(planConfigurationRequest, campaignResponse));
+			log.info("Campaign Integration successful.");
+		} catch (Exception e) {
+			log.error(ServiceConstants.ERROR_WHILE_SEARCHING_CAMPAIGN
+					+ planConfigurationRequest.getPlanConfiguration().getExecutionPlanId(), e);
+			throw new CustomException("Failed to update campaign details", e.toString());
```

Comment:
error code MODULE_FLOW_CLASS_METHOD_ERROR REASON

Reply:
Error code change to "Failed to update campaign details in CampaignIntegration class within method updateCampaignDetails."